src/modules/notification/eventHandlers/onProjectCompletionDueDateSet.ts

```diff
@@ -0,0 +1,73 @@
+import { logger } from '@core/utils'
+import { ProjectRepo, UserRepo } from '@dataAccess'
+import { ProjectCompletionDueDateSet } from '@modules/project'
+import routes from '@routes'
+import { NotificationService } from '../NotificationService'
+
+type OnProjectCompletionDueDateSet = (évènement: ProjectCompletionDueDateSet) => Promise<void>
+
+type MakeOnProjectCompletionDueDateSet = (dépendances: {
+  sendNotification: NotificationService['sendNotification']
+  getProjectUsers: ProjectRepo['getUsers']
+  getProjectById: ProjectRepo['findById']
+  findUsersForDreal: UserRepo['findUsersForDreal']
+}) => OnProjectCompletionDueDateSet
+
+export const makeOnProjectCompletionDueDateSet: MakeOnProjectCompletionDueDateSet =
+  ({ sendNotification, getProjectUsers, getProjectById, findUsersForDreal }) =>
+  async ({ payload: { projectId, reason } }) => {
+    if (reason !== 'délaiCdc2022') {
+      return
+    }
+
+    const projet = await getProjectById(projectId)
+    if (!projet) {
+      logger.error(
+        `Notification handler onProjectCompletionDueDateSet pour délai CDC 2022: projet ${projectId} non trouvé.`
+      )
+      return
+    }
+    const porteursEmails = await getProjectUsers(projectId)
+
+    await Promise.all(
+      porteursEmails.map(({ email, fullName, id }) =>
+        sendNotification({
+          type: 'pp-delai-cdc-2022-appliqué',
+          context: { projetId: projectId, utilisateurId: id },
+          variables: {
+            nom_projet: projet.nomProjet,
+            projet_url: routes.PROJECT_DETAILS(projectId),
+          },
+          message: {
+            email,
+            name: fullName,
+            subject: `Potentiel - Nouveau délai appliqué pour votre projet ${projet.nomProjet}`,
+          },
+        })
+      )
+    )
+
+    const regions = projet.regionProjet.split(' / ')
+    await Promise.all(
+      regions.map(async (region) => {
+        const dreals = await findUsersForDreal(region)
+        Promise.all(
+          dreals.map(({ email, fullName, id }) =>
+            sendNotification({
+              type: 'dreals-delai-cdc-2022-appliqué',
+              context: { projetId: projectId, utilisateurId: id },
+              variables: {
+                nom_projet: projet.nomProjet,
+                projet_url: routes.PROJECT_DETAILS(projectId),
+              },
+              message: {
+                email,
+                name: fullName,
+                subject: `Potentiel - Nouveau délai appliqué pour le projet ${projet.nomProjet}`,
+              },
+            })
+          )
+        )
+      })
+    )
```

Comment:
Ici je crois pas qu'on ait besoin des Promise.all vu qu'on se sert pas des retours de promise.
```suggestion
regions.forEach(async (region) => {
        const dreals = await findUsersForDreal(region)
        dreals.forEach(({ email, fullName, id }) =>
            sendNotification({
              type: 'dreals-delai-cdc-2022-appliqué',
              context: { projetId: projectId, utilisateurId: id },
              variables: {
                nom_projet: projet.nomProjet,
                projet_url: routes.PROJECT_DETAILS(projectId),
              },
              message: {
                email,
                name: fullName,
                subject: `Potentiel - Nouveau délai appliqué pour le projet ${projet.nomProjet}`,
              },
            })
          )
      })
```

Reply:
Il me semble que jusqu'à maintenant on a utilisé un Promise.all pour l'envoi des mails dans tous nos handlers pour exécuter les envois en parallèle et non les uns après les autres. Mais on peut en discuter !

Reply:
Ok, j'approve et ce sera peut-être l'objet d'un clean lors d'une PR dédiée